src/main/kotlin/org/jetbrains/research/testspark/tools/llm/generation/hf/HuggingFaceRequestManager.kt

```diff
@@ -48,22 +49,33 @@ class HuggingFaceRequestManager(project: Project) : IJRequestManager(project) {
 
         val llmRequestBody = HuggingFaceRequestBody(chatHistory, Parameters(topProbability, temperature)).toMap()
         var sendResult = SendResult.OK
+        try {
+            httpRequest.connect {
+                it.write(GsonBuilder().disableHtmlEscaping().create().toJson(llmRequestBody))
+                when (val responseCode = (it.connection as HttpURLConnection).responseCode) {
+                    HttpURLConnection.HTTP_OK -> (testsAssembler as JUnitTestsAssembler).consumeHFRequest(it)
+                    HttpURLConnection.HTTP_INTERNAL_ERROR -> {
+                        llmErrorManager.errorProcess(
+                            LLMMessagesBundle.get("serverProblems"),
+                            project,
+                            errorMonitor,
+                        )
+                        sendResult = SendResult.OTHER
+                    }
 
-        httpRequest.connect {
-            it.write(GsonBuilder().disableHtmlEscaping().create().toJson(llmRequestBody))
-            when (val responseCode = (it.connection as HttpURLConnection).responseCode) {
-                HttpURLConnection.HTTP_OK -> (testsAssembler as JUnitTestsAssembler).consumeHFRequest(it)
-                HttpURLConnection.HTTP_INTERNAL_ERROR -> {
-                    llmErrorManager.errorProcess(
-                        LLMMessagesBundle.get("serverProblems"),
-                        project,
-                        errorMonitor,
-                    )
-                    sendResult = SendResult.OTHER
+                    HttpURLConnection.HTTP_BAD_REQUEST -> {
+                        llmErrorManager.errorProcess(
+                            "The selected model may need an HF PRO subscription to use!",
```

Comment:
could you, please, move this text to the `LLMMessages.properties` and call `LLMMessagesBundle.get(...)` here

Reply:
thank you for fixing